### PR TITLE
Adding Analytics logging for JSRefactor, Live Preview, Quick Edit and more features

### PR DESCRIPTION
**Pull request by sobisht**
This PR is related to adding analytics log for following features in brackets:

1. JSRefactor:

- Rename
- TryCatch
- WrapInCondition
- ConvertToFunction
- CreateGetterSetter
- ExtractToVariable
- ExtractToFunction 

2. QuickEdit: open

3. QuickDoc:
 - Open
 - ReadMore

4. Live Preview: Open

5. Project Settings: Edit in url

6. MultiCursor: Use

ping @swmitra  @navch

### CODE REVIEW

File: src/extensions/default/JavaScriptRefactoring/main.js [hunk 1]
```diff
@@ -66,6 +68,58 @@ define(function (require, exports, module) {
...
+            break;
+        case REFACTORCREATEGETSET:
+            eventName = REFACTORCREATEGETSET;
+            eventType = "createGetterSetter";
+            WrapSelection.createGettersAndSetters();
+            break;
+        }
+        if (eventName) {
+            // Send analytics data for refactoring
+            HealthLogger.sendAnalyticsData(
```
Comment (navch) [inline src/extensions/default/JavaScriptRefactoring/main.js]: We are not even checking whether we are in js context or not? These commands are available in all type of files. Do we just want capture click on the menu or successful usage of this feature?

File: src/extensions/default/JavaScriptQuickEdit/main.js [hunk 2]
```diff
@@ -196,6 +197,14 @@ define(function (require, exports, module) {
             return null;
         }
 
+        //Send analytics data for Quick Edit open
+        HealthLogger.sendAnalyticsData(
```
Comment (navch) [inline src/extensions/default/JavaScriptQuickEdit/main.js]: Again do we are just capturing quick edit performed. Are we interested in errors which we get in this operation?

File: src/extensions/default/MDNDocs/main.js [hunk 3]
```diff
@@ -102,6 +103,15 @@ define(function (require, exports, module) {
             return null;
         }
 
+        // Send analytics data for Quick Doc open
+        HealthLogger.sendAnalyticsData(
+            "cssQuickDoc",
+            "usage",
```
Comment (navch) [inline src/extensions/default/MDNDocs/main.js]: 2nd param is event category? usage might not be the correct word. feature name should be there something like quickedit, multicursor, refactor etc

**Comment by navch** [comment]
Just one more question @sobisht 
What is going to happen, if we are just running tests?

File: src/extensions/default/JavaScriptQuickEdit/main.js (hunk 2 above)
Comment (sobisht) [reply src/extensions/default/JavaScriptQuickEdit/main.js]: we just capture logging when quick edit is performed.

File: src/extensions/default/MDNDocs/main.js (hunk 3 above)
Comment (sobisht) [reply src/extensions/default/MDNDocs/main.js]: @SnchitGrover would be able to clarify this.

File: src/extensions/default/JavaScriptRefactoring/main.js (hunk 1 above)
Comment (sobisht) [reply src/extensions/default/JavaScriptRefactoring/main.js]: Added the check for JS context.

**Comment by sobisht** [comment]
@navch Since unit tests will call the entry functions. Logging will be sent for the unit tests also.

**Review by navch — APPROVED** [review]
Thanks @sobisht for this PR. I have restarted the build.

**Pull request merged**